diff --git a/public/emHdBindings.js b/public/emHdBindings.js
--- a/public/emHdBindings.js
+++ b/public/emHdBindings.js
@@ -11963,4 +11953,6 @@
    Module["FS_unlink"] = FS.unlink;
    Module["PThread"] = PThread;
+    Module["FS_readdir"] = FS.readdir;
+    Module["FS_analyzePath"] = FS.analyzePath;
+    Module["FS_readFile"] = FS.readFile;
+    Module["FS_writeFile"] = FS.writeFile;
     var calledRun;
     dependenciesFulfilled = function runCaller() {
